### PR TITLE
Feature/add auth accounts

### PR DESCRIPTION
**Pull request by ktayah**

### CODE REVIEW

File: lib/belay_brokerage/auth_account.ex [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+      define_field: false
+    )
+  end
+
+  @spec create_changeset(__MODULE__.t(), map()) :: Ecto.Changeset.t()
+  def create_changeset(%__MODULE__{} = struct \\ %__MODULE__{}, params) do
+    struct
+    |> changeset(params)
+    |> unique_constraint([:email, :investor_id])
+    |> foreign_key_constraint(:investor_id)
```
Comment (jeffdeville) [inline lib/belay_brokerage/auth_account.ex:23]: We're going to have to limit is_primary to 1 "true" value

Comment (jeffdeville) [reply lib/belay_brokerage/auth_account.ex:23]: For now, if it's the first entry for a particular investor_id, that'll be fine.

File: lib/belay_brokerage.ex [hunk 2]
```diff
@@ -2,73 +2,83 @@ defmodule BelayBrokerage do
...
-          required(:email) => String.t(),
           required(:phone) => String.t(),
           required(:access_token) => String.t(),
-          required(:account_id) => String.t(),
-          required(:auth0_ids) => [String.t()],
-          required(:item_id) => String.t(),
-          required(:dwolla_customer_id) => String.t()
+          optional(:account_id) => String.t(),
+          optional(:auth_accounts) => [AuthAccount.t()],
+          optional(:item_id) => String.t(),
```
Comment (jeffdeville) [inline lib/belay_brokerage.ex]: I do wish I knew what an item_id was

Comment (ktayah) [reply lib/belay_brokerage.ex]: It's how plaid identifies a "plaid link"

Comment (jeffdeville) [reply lib/belay_brokerage.ex]: Can we name it something plaid-ish? item_id is so generic it's a little hard to follow

Comment (ktayah) [reply lib/belay_brokerage.ex]: Sure, I'll prefix them with plaid
